### PR TITLE
fix(defaults): set proper Accept-Encoding for server side

### PR DESCRIPTION
**Pull request by jericopulvera**
Resolves https://github.com/nuxt-community/axios-module/issues/170

### CODE REVIEW

**Comment by dohomi** [comment]
I had the exact same issue. Would be great being merged

File: lib/plugin.template.js [hunk 1]
```diff
@@ -158,6 +158,7 @@ export default (ctx, inject) => {
     // Axios creates only one which is shared across SSR requests!
     // https://github.com/mzabriskie/axios/blob/master/lib/defaults.js
     headers: {
+      'Accept-Encoding': process.browser ? 'gzip, deflate, br' : 'gzip, deflate',
```
Comment (pi0) [inline lib/plugin.template.js]: This could be better to not change client-side defaults only set it in server side.

Comment (pi0) [reply lib/plugin.template.js]: PS `Accept-Encoding` is forbidden to set in Browsers: 

- https://github.com/axios/axios/issues/542#issuecomment-262146813
- https://xhr.spec.whatwg.org/#the-setrequestheader()-method
- https://stackoverflow.com/questions/22362169

Comment (pi0) [reply lib/plugin.template.js]: PS `Accept-Encoding` is forbidden to set in Browsers: 

- https://github.com/axios/axios/issues/542#issuecomment-262146813
- https://xhr.spec.whatwg.org/#the-setrequestheader()-method
- https://stackoverflow.com/questions/22362169

Comment (manniL) [reply lib/plugin.template.js]: Good catch! That's true. Setting it on client-side shouldn't have an effect at all

Comment (jericopulvera) [reply lib/plugin.template.js]: updated.

File: lib/plugin.template.js [hunk 2]
```diff
@@ -148,6 +148,22 @@ const setupProgress = (axios, ctx) => {
...
+    delete: {},
+    get: {},
+    head: {},
+    post: {},
+    put: {},
+    patch: {}
+  }
+
+  if (process.server) {
+    headers['Accept-Encoding'] = 'gzip, deflate'
```
Comment (manniL) [inline lib/plugin.template.js]: That should be in `common` I think 🤔

**Comment by jericopulvera** [comment]
Thank you!

File: lib/plugin.template.js (hunk 2 above)
Comment (jericopulvera) [reply lib/plugin.template.js]: @manniL @pi0 don't know why but It didn't work for me when it's in common.

tried it locally.

**Pull request merged**